tensorboard/plugins/text/text_plugin.py

```diff
@@ -271,44 +273,52 @@ def _async_index_impl(self):
 
   def index_impl(self):
     run_to_series = self._fetch_run_to_series_from_multiplexer()
+
     # A previous system of collecting and serving text summaries involved
     # storing the tags of text summaries within tensors.json files. See if we
     # are currently using that system. We do not want to drop support for that
     # use case.
     name = 'tensorboard_text'
     run_to_assets = self._multiplexer.PluginAssets(name)
     for run, assets in run_to_assets.items():
+      if run in run_to_series:
+        # When runs conflict, the summaries created via the new method override.
+        continue
+
       if 'tensors.json' in assets:
         tensors_json = self._multiplexer.RetrievePluginAsset(
             run, name, 'tensors.json')
         tensors = json.loads(tensors_json)
-        run_to_series[run] += tensors
+        run_to_series[run] = tensors
       else:
-        run_to_series[run] += []
+        # The mapping should contain all runs among its keys.
```

Comment:
Ah ok, good catch.  I feel like some plugins expect this (that `index_impl()` includes all runs, not just those with plugin data for that plugin) but not all of them?  Do we have a set standard or is this just up to any plugin?

Reply:
For all plugins, the "runs"/"tags" route should list all runs, not just those that include text summaries. Without this, the existing infrastructure breaks when it tries to look up non-plugin-specific runs in the run-to-tag mapping. There's a brief description within the internal change that removes the `tf-sidebar-helper` component.